### PR TITLE
feat(client): add publication content field editing 

### LINKED ISSUE #1118: Add interface to edit publication content fields
### Problem/Why
Publication content fields should be editable from the client interface.

### Goal/What
There should be a page / section of the publication admin pages that allows a publication admin to edit the `home_page_content` and `submission_page_content` fields of a publication.

### Solution/How
Create an interface reachable from admin/publication_details that allows a publication admin to edit the above fields in a QEditor component.

### To Do / Acceptance Criteria

- [ ] Create interface
- [ ] Add jest tests as appropriate
- [ ] Add cypress tests as appropriate

### Review Checklist

- [ ] Content fields are editable
- [ ] Added/modified tests appropriately cover the new feature

### Accessibility Checklist

- [ ] Accessibility testing is performed on new fields / interface.

### PR DESCRIPTION
**Pull request by wreality**
closes #1118

also: 
feat(client): improve publication setup interface
feat(client): separate application admin restricted pages in sidebar menu
feat(api): update unique validation message to be parsable by the client
test(api): move publication content seeding to factory
fix(client): add missing pagination of publication results
refactor(client): avoid side-effects in pagination composable
fix(client): reset client on logout to clean up user data

### CODE REVIEW

**Review by gmeben — CHANGES_REQUESTED** [review]
Since this PR is so large, I'm just going to note things as I find them instead of all at once for the sake of time.

File: client/src/i18n/en-US/index.js [hunk 1]
```diff
@@ -332,9 +389,6 @@ export default {
     create: {
       success: "Publication successfully created.",
       failure: "An error occurred while attempting to create the publication.",
-      required: "A name is required to create a publication.",
-      maxLength: "The maximum length has been exceeded for the name.",
-      duplicate_name: "Publication name already exists.",
     },
     style_criteria: {
       heading: "Style Criteria",
```
[the remark this answers is not in the document]
Comment (wreality) [reply client/src/i18n/en-US/index.js:394]: Yep, found this this morning and have a fix in the works.

**Review by gmeben — CHANGES_REQUESTED** [review]
I encounter an error when attempting to modify style criteria at `/publication/1/setup/criteria`

Screen Shot 2022-06-24 at 3 09 38 PM

**Comment by wreality** [comment]
> I encounter an error when attempting to modify style criteria at `/publication/1/setup/criteria`
> 
> Screen Shot 2022-06-24 at 3 09 38 PM



> I encounter an error when attempting to modify style criteria at `/publication/1/setup/criteria`

Can you please be more specfic.  Which user are you logged in as, when does the error occur (after click, after typing, after submitting?)?  I'm not able to replicate the problem and the tests which edit them are passing.

**Review by gmeben — APPROVED** [review]
Maybe there was some kind of caching issue because I restarted my environment and re-authenticated and I stopped encountering the error I was encountering before.

The steps I took in my replication attempt were:

1. Restart instance
2. Sign in as the application administrator
3. Visit the Publications page
4. Click the first sample publication "CCR Test Publication 1"
5. Click the "Configure Publication" button
6. Select the "Style Criteria" option in the menu
7. Click the pencil icon to modify a style criteria

Earlier, the style criteria would disappear as soon as I clicked the modification icon and page navigation would malfunction. After refreshing the page, the style criteria would return and the page would be reset to normal. 

This error behavior no longer occurs and the rest of the site works as expected as far as I can tell.

**Pull request merged**